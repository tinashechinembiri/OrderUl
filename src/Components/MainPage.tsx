
import WhatsOn from "./HomePage/WhatsOn";

const MainPage = () => 
{
 
        return(
            <div className="mainpage">
            <WhatsOn/>
            </div>   
        );   
    
}
export default MainPage; 