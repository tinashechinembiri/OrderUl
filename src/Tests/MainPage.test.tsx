import React, {ReactNode ,useState as useStateMock }from 'react';
import  MainPage from'../Components/MainPage';
import {create, act} from 'react-test-renderer';
import ReactDOM from 'react-dom';

//https://ogzhanolguncu.com/blog/testing-with-react-testing-library-typescript-and-axios
//https://stackoverflow.com/questions/70400540/dispatch-a-custom-event-and-test-if-it-was-correctly-triggered-react-typescript
describe ('mainpage', () => {
    let SetInput: React.Dispatch<any>
    let SetSearchData:React.Dispatch<any>; 
    const oldCreatePortal = ReactDOM.createPortal;
    beforeAll (()=> {
        ReactDOM.createPortal = (node: ReactNode): React.ReactPortal =>
        node as React.ReactPortal;
    })
    afterAll(() => {
        ReactDOM.createPortal = oldCreatePortal;
      });
      beforeEach(()=> {
        SetInput = jest.fn(x=> {});
        SetSearchData = jest.fn(x=> {});

       React.useState = jest.fn().mockImplementationOnce ( x => [x,SetInput])
                                  .mockImplementationOnce(x => [x, SetSearchData])
    })
    it('create a mainpage snapshot', async() => {
        await act (async () => {
            const tree = await create(<MainPage/>)
            expect (tree).toMatchSnapshot(); 
        })
    }); 

    it ('create a snap shot after', () => {
        const tree =  create(<MainPage/>)
        expect (tree).toMatchSnapshot(); 
    })
    
    
    
})