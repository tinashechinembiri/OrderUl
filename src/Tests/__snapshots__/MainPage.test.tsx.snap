// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainpage create a mainpage snapshot 1`] = `
<div
  className="mainpage"
>
  <div>
    Loading...
  </div>
</div>
`;

exports[`mainpage create a snap shot after 1`] = `
<div
  className="mainpage"
>
  <div
    className="input-bars"
  >
    <div
      className="searchbars-total"
    >
      <div
        aria-level={2}
        className="display-header"
        role="heading"
      >
         Whats on
      </div>
      <div
        className="movie-input"
      >
        <input
          className="location-input"
          onClick={[Function]}
          readOnly={true}
          value="choose your location"
        />
        <input
          className="search_input"
          id="movie-search"
          onChange={[Function]}
          placeholder="Start typing a film or event"
          type="text"
          value=""
        />
      </div>
    </div>
    <ul
      className="ul-list-adverts"
    />
  </div>
</div>
`;
